src/EntityFramework/Migrations/Infrastructure/EdmModelDiffer.cs

```diff
@@ -328,6 +328,27 @@ where m1.EdmEquals(m2)
                     / (entityType1.DeclaredMembers.Count + entityType2.DeclaredMembers.Count)) > 0.80;
         }
 
+        private bool SourceAndTargetMatch(EntityType sourceEntityType, EntityTypeMapping sourceEntityTypeMapping, EntityType targetEntityType, EntityTypeMapping targetEntityTypeMapping)
+        {
+            return (sourceEntityTypeMapping.EntityType != null
+                    && targetEntityTypeMapping.EntityType != null
+                    && sourceEntityType == sourceEntityTypeMapping.EntityType
+                    && targetEntityType == targetEntityTypeMapping.EntityType)
+                   || (sourceEntityTypeMapping.EntityType == null
+                       && targetEntityTypeMapping.EntityType == null
+                       && sourceEntityTypeMapping.IsOfTypes.Contains(sourceEntityType)
```

Comment:
Follow up... looks like it's doing this for every call :(
`return new ReadOnlyCollection<EntityTypeBase>(new List<EntityTypeBase>(m_isOfEntityTypes.Values));`

So that's bad.